### DISCUSSION TITLE
SEO and server side rendering

### DISCUSSION THREAD

**Discussion by geekyme-fsmk** [SAP/luigi#2670] [Q&A] [answered]
This is a pretty cool framework. I'm wondering about these concerns:

- How SEO-ready is a web app made with luigi? Is the content inside each microfrontend indexable?
- Can luigi control the load order for different microfrontends rendered on the server side? Example: imagine facebook built with luigi, with 3 separate microfrontend: news feed, settings, messenger. News feed should be loaded up first, followed by messenger and finally settings.

**Comment by ndricimrr** [comment, marked as answer]
Hi @geekyme-fsmk , thanks for your question.

> How SEO-ready is a web app made with luigi? Is the content inside each microfrontend indexable?

Luigi relies on **iFrames** and **Web Components** as the two main technologies used for building the microfrontends.
SEO-readiness of an app built with Luigi then depends on the indexability of the iFrames or Web Components. As an example, our [documentation website](https://docs.luigi-project.io/docs/getting-started) itself is built with Luigi and it internally relies on iFrames where each documentation section is a separate microfrontend.

> Can luigi control the load order for different microfrontends rendered on the server side? Example: imagine facebook built with luigi, with 3 separate microfrontend: news feed, settings, messenger. News feed should be loaded up first, followed by messenger and finally settings.

Loading several different microfrontends in one page is possible with Luigi. We use [Web Components](https://docs.luigi-project.io/docs/faq/?section=can-i-place-a-micro-frontend-within-another-micro-frontend) to offer this feature. When having multiple such microfrontends in one page, they are loaded sequentially according to their order on the configuration file. It is however possible that we might implement a customisable rendering order feature, should there be demand for it.
